.github/workflows/c-cpp.yml

```diff
@@ -0,0 +1,60 @@
+name: C/C++ CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        config: [Release]
+
+    steps:
+      - name: Set up build environment (macos-latest)
+        run: |
+          brew install ccache boost
+          echo "::add-path::/usr/local/opt/ccache/libexec"
+          echo "::set-env name=CCACHE_DIR::/tmp/ccache"
+        if: matrix.os == 'macos-latest'
+
+      - name: Set up build environment (ubuntu-latest)
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install ccache libboost-all-dev libgtk-3-dev libsdl2-dev
+          echo "::set-env name=CCACHE_DIR::/tmp/ccache"
+        if: matrix.os == 'ubuntu-latest'
+
+      - name: Set up build environment (windows-latest)
+        run: |
+          echo "::set-env name=BOOST_ROOT::$env:BOOST_ROOT_1_72_0"
+        if: matrix.os == 'windows-latest'
+
+      - uses: actions/cache@v1
+        with:
+          path: /tmp/ccache
+          key: ccache-${{ matrix.os }}-${{ matrix.config }}-${{ github.sha }}
+          restore-keys: ccache-${{ matrix.os }}-${{ matrix.config }}-
+        if: matrix.os != 'windows-latest'
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: CMake configure
+        run: cmake -B build -DCI=ON -DCMAKE_BUILD_TYPE=${{ matrix.config }}
+
+      - name: CMake build
+        run: cmake --build build --config ${{ matrix.config }} --parallel 2
```

Comment:
Why not use `j` instead of `parallel`?

Reply:
Does Github Action have cross-platforms `nproc`? This will reduce build time significantly.

Reply:
Now, I know why those builds took forever to complete on CI.

Reply:
@Margen67 `parallel` and `j` are the same. Source: https://cmake.org/cmake/help/latest/manual/cmake.1.html

@ichisadashioko As far as I know, GitHub Actions don't have cross-platform `nproc`. I an using the value "2" because it is the number of cpu available. Source: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources

Reply:
> 
> 
> @Margen67 `parallel` and `j` are the same. Source: https://cmake.org/cmake/help/latest/manual/cmake.1.html
> 
> @ichisadashioko As far as I know, GitHub Actions don't have cross-platform `nproc`. I am using the value "2" because it is the number of cpu available. Source: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources

For Windows there's an environment variable: `NUMBER_OF_PROCESSORS`
For macOS you can do use this (or a similar) command: `sysctl hw.ncpu`

Reply:
Yeah, but Github Action uses the same command for all OSs.